lib/src/validationEngine.ts

```diff
@@ -71,61 +73,59 @@ export class ValidationEngine implements IValidationEngine {
         })
         .catch((result) => {
           // Build failed validation Result
-          var errorInformation = `Uncontrolled error when validating full form, check custom validations code`;
+          const errorInformation = `Uncontrolled error when validating full form, check custom validations code`;
           console.log(errorInformation);
-          reject(errorInformation);
+          reject(result);
         });
-
     });
 
     return fullFormValidatedPromise;
   }
 
-  triggerFieldValidation(vm: any, key: string, value: any, filter: any = consts.defaultFilter): Promise<FieldValidationResult> {
+  triggerFieldValidation(vm: any, key: string, value: any, filter: ValidationFilters = consts.defaultFilter): Promise<FieldValidationResult> {
```

Comment:
customParams should be added when you declare your field validation constraint. It is supplied by the validationDispatcher when fires the validation function. Should it be supplied by the action / whatever needs the validation result in runtime?